### PR TITLE
docs: update contributing guide by adding upstream remote

### PR DESCRIPTION
**Pull request by reshamas**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #1508 
- Closes #\<issue number\>

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

**Comment by reshamas** [comment]
Note: this is at at the PyData Global online sprint.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -44,6 +44,28 @@ for example:
 git clone git@github.com:YOUR-USERNAME/narwhals.git
 ```
 
+### 3. Add the `upstream` remote
```
Comment (MarcoGorelli) [inline CONTRIBUTING.md:47]: like this we've got two `### 3.`, shall we update the numbers?

Comment (FBruzzesi) [inline CONTRIBUTING.md:47]: Should all the numbers be updated? Or this could be a sub-section within 3 ✨

**Review by FBruzzesi — COMMENTED** [review]
Thanks for spotting and fixing this! I left a tiny comment in the enumeration 🤞🏼

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -44,6 +44,28 @@ for example:
...
+git remote -v
+```
+
+for example:  
+```bash
+git remote -v                                                          
+origin	git@github.com:ghusername/narwhals.git (fetch)
+origin	git@github.com:ghusername/narwhals.git (push)
+upstream	https://github.com/narwhals-dev/narwhals.git (fetch)
+upstream	https://github.com/narwhals-dev/narwhals.git (push)
```
Comment (MarcoGorelli) [inline CONTRIBUTING.md:65]: like this, one shows ssh and the other https

can we show `git@github.com:narwhals-dev/narwhals.git` for upstream?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -44,6 +44,28 @@ for example:
...
+Check to see the remote has been added: 
+
+```bash
+git remote -v
+```
+
+for example:  
+```bash
+git remote -v                                                          
+origin	git@github.com:ghusername/narwhals.git (fetch)
```
Comment (MarcoGorelli) [inline CONTRIBUTING.md:62]: how about
```
origin	git@github.com:<your GitHub username>/narwhals.git (fetch)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -44,6 +44,28 @@ for example:
...
+git remote add upstream git@github.com:narwhals-dev/narwhals.git
+``` 
+
+Check to see the remote has been added: 
+
+```bash
+git remote -v
+```
+
+for example:
```
Comment (MarcoGorelli) [inline CONTRIBUTING.md:59]:
```suggestion
You should see something like this:  
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -44,6 +44,28 @@ for example:
 git clone git@github.com:YOUR-USERNAME/narwhals.git
 ```
 
+### 3. Add the `upstream` remote
+
+```bash
+git remote add upstream git@github.com:narwhals-dev/narwhals.git
```
Comment (MarcoGorelli) [inline CONTRIBUTING.md:50]: can we also add `git fetch upstream` underneath?

**Review by MarcoGorelli — COMMENTED** [review]
awesome, thanks! good to have you as contributor

File: CONTRIBUTING.md (hunk 1 above)
Comment (reshamas) [reply CONTRIBUTING.md]: Updated. thank you.

Comment (reshamas) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (reshamas) [reply CONTRIBUTING.md:62]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (reshamas) [reply CONTRIBUTING.md:63]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (reshamas) [reply CONTRIBUTING.md]: done

**Review by MarcoGorelli — APPROVED** [review]
thanks @reshamas !

**Pull request merged**